### PR TITLE
feat: Cleanup and reorganize VPP section

### PR DESCRIPTION
**Pull request by jimmarino**
## WHAT

Cleans up and reorganizes the Verifiable Presentation Protocol Section:

1. Moves terms to the terminology section and improves the definitions
2. Enhances the VPP flow diagram to include DID resolution
3. Improve VPP flow description
4. Moves scope section to the Resolution API where it is used
5. Removes `read | write | all` and wildcards from the scope definition as they are no longer used
6. Add references to terms and definitions
7. Add normative language

### CODE REVIEW

File: specifications/terminology.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
 # Terminology
```
Comment (nikschul) [inline specifications/terminology.md:2]: Currently we define our terminology over multiple documents, e.g. also in the CIP and VPP specifications. What about removing the Terms section in all other documents and just referring to this markdown file instead?

File: specifications/terminology.md [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
 
-The following terms are used to describe concepts in this specification.
\ No newline at end of file
+The following terms are used to describe concepts in this specification.
+
+- <dfn>DID</dfn> - A decentralized identifier as defined by [[[did-core]]].
+- <dfn>Holder</dfn> - An entity that possesses a set of identity resources as defined by [[[vc-data-model]]].
+  The holder will typically be the subject of a [=Verifiable Credential=].
+- <dfn>Resource</dfn> - A resource is an entity managed by the Credential Service such as a [=Verifiable Credential=]
```
Comment (nikschul) [inline specifications/terminology.md:8]: Maybe the definition of "Credential Service" would also be nice to have here, which I proposed in my PR #91.
```
- **Credential Service** - The CS is a network-accessible service that manages Verifiable Credentials [[vc-data-model]]. This includes read and write endpoints
  for Verifiable Presentations (VPs) and Verifiable Credentials (VCs).
```

File: specifications/terminology.md (hunk 1 above)
Comment (jimmarino) [reply specifications/terminology.md:2]: Yes, that's what I'm doing in steps. This PR moved them from the VPP section. The next PR will move any remaining terms from the CIP section.

File: specifications/terminology.md (hunk 2 above)
Comment (jimmarino) [reply specifications/terminology.md:8]: The CS is defined in the ecosystem section. Let's decide whether to keep it there, merge the sections, or duplicate definitions in a subsequent PR.

**Review by nikschul — APPROVED** [review]

**Review by arnoweiss — COMMENTED** [review]
No objections. Will start work on #96 when merged.

**Review by arnoweiss — APPROVED** [review]

**Review by paullatzelsperger — APPROVED** [review]

**Review by wolf4ood — APPROVED** [review]

**Pull request merged**